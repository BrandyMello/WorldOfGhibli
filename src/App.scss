main {
  font-family: 'Gayathri', sans-serif;
  font-weight: bold;
  color: white;
  overflow: scroll;
  background: url("http://41.media.tumblr.com/b4f3eb1230bebbc4d39964dfda9159f3/tumblr_nic5r95Fam1qcopb3o1_1280.png") no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}

h1 {
  position: fixed;
  top: 79px;
  left: 0;
  writing-mode: vertical-lr;
  text-orientation: upright;
  height: 100%;
  background-color: #000000db;
  padding: 15px;
  z-index: 5;
}

nav,
footer {
  height: 60px;
  background-color: black;
}

nav {
  position: fixed;
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  z-index: 2;
}

input,
button {
  font-family: 'Gayathri', sans-serif;
  font-weight: bold;
  height: 30px;
  margin: 10px;
}

button {
  width: 60px;
}

form {
  display: flex;
  align-items: center;
}

input {
  font-size: 15px;
  width: 220px;
}