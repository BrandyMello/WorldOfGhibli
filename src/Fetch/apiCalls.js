export const getFilms = () => {
  return fetch('https://ghibliapi.herokuapp.com/films/')
    .then(response => {
      if (!response.ok) {
        throw Error('Error fetching films')
      }
      return response.json()
    })
    .catch(error => {
      throw Error(error.message)
    })
}

export const getCharacters = () => {
  return fetch('https://ghibliapi.herokuapp.com/people/')
    .then(response => {
      if(!response.ok) {
        throw Error('Error fetching characters')
      }
      return response.json()
    })
    .catch(error => {
        throw Error(error.message)
    })  
}
