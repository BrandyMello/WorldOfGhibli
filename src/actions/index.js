export const storeFilms = (films) => ({
  type: 'STORE_FILMS',
  films
});

export const storeCharacters = (characters) => ({
  type: 'STORE_CHARACTERS',
  characters
});

export const storeCharNames = (characterNames) => ({
  type: 'STORE_CHAR_NAMES',
  characterNames
});

export const setCurrentPlayer = (playerName) => ({
  type: 'SET_CURRENT_PLAYER',
  playerName
});

export const setWins = (numOfWins) => ({
  type: 'SET_WINS',
  numOfWins
})