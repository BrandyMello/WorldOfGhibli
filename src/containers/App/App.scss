main {
  color: white;
  font-family: 'Gayathri', sans-serif;
  font-weight: bold;
  z-index: -1;
}

.section_films_display {
  overflow-y: scroll;
  background: url("http://41.media.tumblr.com/b4f3eb1230bebbc4d39964dfda9159f3/tumblr_nic5r95Fam1qcopb3o1_1280.png") no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}

.section_game_display {
  background: url(https://cdn.suwalls.com/wallpapers/anime/studio-ghibli-characters-36913-3840x2160.jpg) no-repeat center center fixed;
}

h1 {
  background-color: #000000db;
  height: 100%;
  padding: 15px;
  position: fixed;
  top: 79px;
  left: 0;
  text-orientation: upright;
  writing-mode: vertical-lr;
  z-index: 5;
}

nav,
footer {
  background-color: black;
  height: 60px;
  width: 100%;
}

footer {
  bottom: 0;
  position: fixed;
}

nav {
  align-items: center;
  display: flex;
  justify-content: space-around;
  position: fixed;
  z-index: 2;
}

input,
button {
  height: 30.9px;
  font-family: 'Gayathri', sans-serif;
  font-weight: bold;
  padding-top: 5px;
}

.btn_home {
  border-radius: 3px;
  width: 76px;
}

.input {
  border: none;
  height: 24px;
  border-radius: 3px 0px 0px 3px;
  padding-left: 15px;
}

.btn_play {
  border-radius: 0px 3px 3px 0px;
}

form {
  align-items: center;
  display: flex;
}

input {
  font-size: 15px;
  width: 220px;
}

.arrow {
  position: absolute;
  bottom: 34%;
  left: 97%;
  height: 2em;
  width: 2em;
}