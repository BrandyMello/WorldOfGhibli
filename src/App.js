import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import * as apiCalls from './Fetch/apiCalls.js';
import GameDisplay from './containers/GameDisplay/GameDisplay';
import './App.scss';
import FilmsDisplay from './containers/FilmsDisplay/FilmsDisplay';
import { connect } from 'react-redux';
import { storeFilms, storeCharacters, storeCharNames, setCurrentPlayer } from './actions';
import { bindActionCreators } from 'redux';
// import PropTypes from 'prop-types';

class App extends Component {
  constructor() {
    super();
    this.state={
      player: ''
    }
  }

  handlePlayerInput = (e) => {
    this.setState({[e.target.name]: e.target.value});
  }

  declarePlayer = (e) => {
    e.preventDefault();
    this.props.setCurrentPlayer(this.state.player);
  }

  componentDidMount = () => {
    apiCalls.getFilms()
      .then(films => this.props.storeFilms(films))
      .catch(err => console.error(err));
    
    apiCalls.getCharacters()
      .then(characters => this.props.storeCharacters(characters))
      .catch(err => console.error(err));

    apiCalls.getCharacters()
      .then(characters => this.props.storeCharNames(characters))
      .catch(err => console.error(err));
  }
    
  render() {
    return (
      <main>
        <nav className="nav">
          <NavLink to='/' className="home-nav"><button>HOME</button></NavLink>
          <form className="enter-game">
            <input placeholder="player's name: " name="player" value={this.state.player} onChange={this.handlePlayerInput}></input>
            <NavLink to='/' onClick={this.declarePlayer}>
              <NavLink to='/game' className='game-nav'><button>PLAY GAME</button></NavLink>
            </NavLink>
          </form>
          <h1>World of Ghibli</h1>  
        </nav>
        <Route exact path='/' render={() => <FilmsDisplay films={this.props.films}/>} />
        <Route exact path='/game' render={() => <GameDisplay characters={this.props.characters} player={this.props.player}/>} /> 
        <footer></footer>
      </main> 
    )
  }
}

export const mapStateToProps = state => ({
  films: state.films,
  characters: state.characters,
  characterNames: state.characterNames,
  player: state.player
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({ storeFilms, storeCharacters, storeCharNames, setCurrentPlayer }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(App);


