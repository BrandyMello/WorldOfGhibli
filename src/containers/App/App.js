import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import { getFilms, getCharacters } from '../../Fetch/apiCalls.js';
import GameDisplay from '../../components/GameDisplay/GameDisplay';
import './App.scss';
import FilmsDisplay from '../../components/FilmsDisplay/FilmsDisplay';
import { connect } from 'react-redux';
import { storeFilms, storeCharacters, storeCharNames, setCurrentPlayer } from '../../actions';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { IoIosArrowForward } from 'react-icons/io';

export class App extends Component {
  constructor() {
    super();
    this.state={
      player: ''
    }
  }

  handlePlayerInput = (e) => {
    this.setState({[e.target.name]: e.target.value});
  }

  declarePlayer = (e) => {
    e.preventDefault();
    this.props.setCurrentPlayer(this.state.player);
    this.clearInput();
  }

  clearInput = () => {
    this.setState({player: ''})
  }

  componentDidMount = () => {
    getFilms()
      .then(films => this.props.storeFilms(films))
      .catch(err => console.error(err));
    
    getCharacters()
      .then(characters => this.props.storeCharacters(characters))
      .catch(err => console.error(err));

    getCharacters()
      .then(characters => this.props.storeCharNames(characters))
      .catch(err => console.error(err));
  }
    
  render() {
    return (
      <main>
        <nav className="nav">
          <NavLink to="/" className="home-nav">
            <button class="btn_home">HOME</button>
          </NavLink>
          <form className="enter-game">
            <input class="input"
              placeholder="player's name: "
              name="player"
              value={this.state.player}
              onChange={this.handlePlayerInput}
            ></input>
            <NavLink to="/" onClick={this.declarePlayer}>
              <NavLink to="/game" className="game-nav">
                <button disabled={!this.state.player} class="btn_play">
                  PLAY GAME
                </button>
              </NavLink>
            </NavLink>
          </form>
          <h1>World of Ghibli</h1>
        </nav>
        <Route
          exact
          path="/"
          render={() => (
            <section class="section_films_display">
              <FilmsDisplay films={this.props.films} />
              <IoIosArrowForward class="arrow"/>
            </section>
          )}
        />
        <Route
          exact
          path="/game"
          render={() => (
            <section class="section_game_display">
              <GameDisplay
                characters={this.props.characters}
                player={this.props.player}
              />
            </section>
          )}
        />
        <footer></footer>
      </main>
    );
  }
}

export const mapStateToProps = state => ({
  films: state.films,
  characters: state.characters,
  characterNames: state.characterNames,
  player: [state.player]
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({ storeFilms, storeCharacters, storeCharNames, setCurrentPlayer }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
  films: PropTypes.array.isRequired,
  characters: PropTypes.array.isRequired,
  characterNames: PropTypes.array.isRequired,
  player: PropTypes.array,
  storeFilms: PropTypes.func.isRequired,
  storeCharacters: PropTypes.func.isRequired,
  storeCharNames: PropTypes.func.isRequired,
  setCurrentPlayer: PropTypes.func.isRequired
}
