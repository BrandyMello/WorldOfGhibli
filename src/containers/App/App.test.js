import React from 'react';
import {App, mapStateToProps, mapDispatchToProps} from './App';
import { shallow } from 'enzyme';
import { getFilms, getCharacters } from '../../Fetch/apiCalls';
import { storeFilms, storeCharacters, storeCharNames, setCurrentPlayer } from '../../actions';

jest.mock('../../Fetch/apiCalls');

const mockFilm = {
  id: "2b",
  title: "Castle in the Sky",
  description: "The orphan Sheeta inherited a mysterious...",
  director: "Hayao Miyazaki",
  producer: "Isao Takahata",
  release_date: "1986",
  rt_score: "95",
  people: [
    "https://ghibliapi.herokuapp.com/people/"
  ],
  species: [
    "https://ghibliapi.herokuapp.com/species/af3910a6-429f-4c74-9ad5-dfe1c4aa04f2"
  ],
  locations: [
    "https://ghibliapi.herokuapp.com/locations/"
  ],
  vehicles: [
    "https://ghibliapi.herokuapp.com/vehicles/"
  ],
  url: "https://ghibliapi.herokuapp.com/films/2baf70d1-42bb-4437-b551-e5fed5a87abe"
}

const mockCharacter = {
  id: "ba9",
  name: "Ashitaka",
  gender: "Male",
  age: "late teens",
  eye_color: "Brown",
  hair_color: "Brown",
  films: [
    "https://ghibliapi.herokuapp.com/films/0440483e-ca0e-4120-8c50-4c8cd9b965d6"
  ],
  species: "https://ghibliapi.herokuapp.com/species/af3910a6-429f-4c74-9ad5-dfe1c4aa04f2",
  url: "https://ghibliapi.herokuapp.com/people/ba924631-068e-4436-b6de-f3283fa848f0"
}

getFilms.mockImplementation(() => {
    return Promise.resolve({
      films: [mockFilm, mockFilm]
    })
  });

getCharacters.mockImplementation(() => {
  return Promise.resolve({
    characters: [mockCharacter, mockCharacter]
  })
});

describe('App', () => {
  let wrapper;
  const storeFilmsMock = jest.fn();
  const storeCharactersMock = jest.fn();
  const storeCharNamesMock = jest.fn();
  const setCurrentPlayerMock = jest.fn();
  const mockFilms = [mockFilm, mockFilm];
  const mockCharacters = [mockCharacter, mockCharacter];
  const mockCharacterNames = ['Brandy', 'Fred']
  const mockPlayer = ['Brandy']
  beforeEach(() => {
    wrapper = shallow(<App 
      films={mockFilms}
      characters={mockCharacters}
      characterNames={mockCharacterNames}
      player={mockPlayer}
      storeFilms={storeFilmsMock}
      storeCharacters={storeCharactersMock}
      storeCharNames={storeCharNamesMock}
      setCurrentPlayer={setCurrentPlayerMock}
      getFilms={jest.fn()}
      getCharacters={jest.fn()}
    />)
  });

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change the state on the input change within the form', () => {
        wrapper.find('input').at(0).simulate('change', {target: {value: 'Brandy', name: 'player'}});

        expect(wrapper.state('player')).toEqual('Brandy');
  });

  // it('should invoke declarePlayer when the play game button is clicked', () => {
  //   const mockDelcarePlayer = jest.fn({ preventDefault: jest.fn() });
  //   wrapper.instance().forceUpdate();
  //   wrapper.find('button').at(1).simulate('click');
  //   wrapper.instance().mockDelcarePlayer({ preventDefault: jest.fn() })
  //   expect(mockDelcarePlayer).toHaveBeenCalled();
  // });


  it('should call getFilms and getCharacters after mounting', () => {
    expect(getFilms).toHaveBeenCalled();
    expect(getCharacters).toHaveBeenCalled();
  })

  it('should save the player input name to state', () => {
    wrapper.find('input').simulate('change', {target: {value: 'Brandy', name: 'player'}});

    expect(wrapper.state('player')).toEqual('Brandy');
  });

  it('should clear the player input when clearInputs is invoked', () => {
    wrapper.instance().clearInputs;
    expect(wrapper.find('input').at(0)).toEqual({});
  });

  it('mapStateToProps should get the props', () => {
    const expectedProps = {
      films: mockFilms,
      characters: mockCharacters,
      characterNames: mockCharacterNames,
      player: mockPlayer
    }
     const mockedProps = {
       films: mockFilms,
       characters: mockCharacters,
       characterNames: mockCharacterNames,
       player: 'Brandy'
     }
    
    const mappedProps = mapStateToProps(mockedProps);
    expect(mappedProps).toEqual(expectedProps);
  });

  describe('mapStateToProps', () => {
    it('should return an array of film objects, an array of character objects, an array of character names, and a player name', () => {
      const mockState = {
        films: [{film1: 'this'}, {film2: 'that'}],
        characters: [{ character1: 'this' }, { character2: 'that' }],
        characterNames: ['Bill', 'Bob'],
        player: 'Brandy'
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
    })
  });

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mappedDispatchedProps = mapDispatchToProps(mockDispatch);

    it('should call dispatch with the storeFilms action', () => {
      const mockFilms = [{ film1: 'stuff' }, { film2: 'stuff' }];
      const dispatchStoreFilmsAction = storeFilms(mockFilms);

      mappedDispatchedProps.storeFilms(mockFilms);

      expect(mockDispatch).toHaveBeenCalledWith(dispatchStoreFilmsAction)
    });

    it('should call dispatch with the storeCharacters action', () => {
      const mockCharacters = [{ character1: 'stuff' }, { character2: 'stuff' }];
      const dispatchStoreCharactersAction = storeCharacters(mockCharacters);

      mappedDispatchedProps.storeCharacters(mockCharacters);

      expect(mockDispatch).toHaveBeenCalledWith(dispatchStoreCharactersAction)
    });

    it('should call dispatch with the storeCharNames action', () => {
      const mockCharacters = [{ character1: 'stuff' }, { character2: 'stuff' }];
      const dispatchStoreCharNamesAction = storeCharNames(mockCharacters);

      mappedDispatchedProps.storeCharNames(mockCharacters);

      expect(mockDispatch).toHaveBeenCalledWith(dispatchStoreCharNamesAction);
    });

    it('should call dispatch with setCurrentPlayer action', () => {
      const mockPlayer = ['Brandy'];
      const dispatchSetCurrentPlayerAction = setCurrentPlayer(mockPlayer);

      mappedDispatchedProps.setCurrentPlayer(mockPlayer);

      expect(mockDispatch).toHaveBeenCalledWith(dispatchSetCurrentPlayerAction);
    });
  });
  
});


