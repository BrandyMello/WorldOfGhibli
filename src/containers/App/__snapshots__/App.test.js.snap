// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with the data passed through 1`] = `
<main>
  <nav
    className="nav"
  >
    <NavLink
      className="home-nav"
      to="/"
    >
      <button>
        HOME
      </button>
    </NavLink>
    <form
      className="enter-game"
    >
      <input
        name="player"
        onChange={[Function]}
        placeholder="player's name: "
        value=""
      />
      <NavLink
        onClick={[Function]}
        to="/"
      >
        <NavLink
          className="game-nav"
          to="/game"
        >
          <button>
            PLAY GAME
          </button>
        </NavLink>
      </NavLink>
    </form>
    <h1>
      World of Ghibli
    </h1>
  </nav>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/game"
    render={[Function]}
  />
  <footer />
</main>
`;
